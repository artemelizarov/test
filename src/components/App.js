import React, { Component } from "react";
import CurrenciesList from "../containers/CurrenciesList";
import CurrencyConverter from "../containers/CurrencyConverter";

class App extends Component {
  componentDidMount() {
    this.props.fetchCurrencies();
    this.props.fetchDefaultCurrency();
  }

  render() {
    const { defaultCurrency, currencies } = this.props;

    if (!defaultCurrency || !currencies.length) {
      return <div className="App">Loading...</div>;
    } else {
      return (
        <div className="App">
          <CurrencyConverter />
          <CurrenciesList />
        </div>
      );
    }
  }
}

export default App;
