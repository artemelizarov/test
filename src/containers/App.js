import { connect } from "react-redux";
import { fetchCurrencies, fetchDefaultCurrency } from "../redux/actions";
import App from "../components/App";

const mapStateToProps = state => ({
  currencies: state.currencies,
  defaultCurrency: state.defaultCurrency
});

const mapDispatchToProps = dispatch => ({
  fetchCurrencies: () => dispatch(fetchCurrencies()),
  fetchDefaultCurrency: () => dispatch(fetchDefaultCurrency())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
