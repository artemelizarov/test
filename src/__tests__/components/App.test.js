// containers, components
import App from "../../containers/App";

describe("render all app", () => {
  let wrapper;

  beforeAll(() => {
    mockAPICalls();

    wrapper = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  test("component is render", () => {
    expect(wrapper.find(".App").length).toBe(1);
  });

  test("CurrencyConverter subcomponent is render", () => {
    wrapper.update();
    expect(wrapper.find(".CurrencyConverter").length).toBe(1);
  });
});

describe("snapshot-test App component", () => {
  test("Renders correct properties", () => {
    shallowExpect(
      <Provider store={store}>
        <App />
      </Provider>
    ).toMatchSnapshot();
  });
});
