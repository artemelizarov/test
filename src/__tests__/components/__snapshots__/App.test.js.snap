// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test App component Renders correct properties 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "currencies": Array [
          Object {
            "code": "EUR",
            "favorite": false,
            "name": "Euro",
            "rate": 100,
          },
          Object {
            "code": "USD",
            "favorite": false,
            "name": "United States Dollar",
            "rate": undefined,
          },
        ],
        "currenciesList": Object {
          "baseCurrency": "USD",
          "currenciesRates": Object {
            "EUR": 100,
            "USD": 50,
          },
          "errorRates": null,
          "favorites": Array [],
        },
        "currencyConverter": Object {
          "amountFrom": 1,
          "amountTo": 100,
          "currencyFrom": "USD",
          "currencyTo": "EUR",
          "rate": 100,
        },
        "defaultCurrency": "USD",
      },
    }
  }
>
  <Connect(App) />
</ContextProvider>
`;
